### PR TITLE
Delete global modal

### PR DESCRIPTION
**Pull request by vadim11041996**

### CODE REVIEW

File: src/app/modules/teachers/containers/teacher-details-page/teacher-details-page.component.ts [hunk 1]
```diff
@@ -42,11 +42,15 @@ export class TeacherDetailsPageComponent implements OnInit, OnDestroy {
     const initialState: any = {
       teacher: this.teacher
```
Comment (vtamazlicari) [inline src/app/modules/teachers/containers/teacher-details-page/teacher-details-page.component.ts]: where ;

File: src/app/modules/teachers/containers/teacher-details-page/teacher-details-page.component.ts [hunk 2]
```diff
@@ -42,11 +42,15 @@ export class TeacherDetailsPageComponent implements OnInit, OnDestroy {
...
     };
-    this.modalEditRef = this.modalService.show(EditTeacherModalComponent, { initialState });
+    this.modal = this.modalService.show(EditTeacherModalComponent, { initialState });
   }
 
   openDeleteModal() {
-    this.modalEditRef = this.modalService.show(DeleteTeacherModalComponent);
+    const initialState = {
+      service: this.teachersService,
+      id: this.id
```
Comment (vtamazlicari) [inline src/app/modules/teachers/containers/teacher-details-page/teacher-details-page.component.ts]: ; ?

File: src/app/modules/teachers/teachers.service.ts [hunk 3]
```diff
@@ -13,7 +13,11 @@ import { TOKEN_STORAGE_KEY } from '@shared/constants/index';
...
 export class TeachersService {
-  id: number;
+
+  itemName: String;
+
+  pageUrl: String = 'teachers';
+
   options = {
     headers: new HttpHeaders({
       "token": this.localStorage.getLocalStorage(TOKEN_STORAGE_KEY)
```
Comment (vtamazlicari) [inline src/app/modules/teachers/teachers.service.ts]: delete header

File: src/app/modules/teachers/teachers.service.ts [hunk 4]
```diff
@@ -25,32 +29,31 @@ export class TeachersService {
...
-    return this.api.post(`teachers`, form)
+    return this.api.post(`${this.pageUrl}`, form)
       .pipe(map(result => result));
   }
 
   editTeacher(form: Teacher): Observable<any> {
-    return this.api.put(`teachers/${form.id}`, form)
+    return this.api.put(`${this.pageUrl}/${form.id}`, form)
       .pipe(first());
   }
```
Comment (vtamazlicari) [inline src/app/modules/teachers/teachers.service.ts]: creates in shared service for admin student teacher

File: src/app/shared/shared.module.ts [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import { NgModule } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { DeleteModalComponent } from './components/delete-modal/delete-modal.component';
```
Comment (vtamazlicari) [inline src/app/shared/shared.module.ts]: use global import

File: src/app/shared/components/confirm-modal/confirm-modal.component.html [hunk 6]
```diff
@@ -1,6 +1,6 @@
 <div>
```
Comment (crischiucucu) [inline src/app/shared/components/confirm-modal/confirm-modal.component.html]: Do you really need this div element?

File: src/app/shared/components/confirm-modal/confirm-modal.component.html [hunk 7]
```diff
@@ -1,6 +1,6 @@
...
   <div class="modal-body text-center">
-    <p>Are you sure, you want to delete this user ?</p>
+    <p>Are you sure ?</p>
```
Comment (crischiucucu) [inline src/app/shared/components/confirm-modal/confirm-modal.component.html]: Here you could includ an @input parameter for customizing the modal text to be shown

File: src/app/shared/components/confirm-modal/confirm-modal.component.html [hunk 8]
```diff
@@ -9,6 +9,6 @@
             (click)="decline()">No</button>
   </div>
   <pre class="card card-block card-header">
-        {{message}}
-  </pre>
+          {{message}}
```
Comment (crischiucucu) [inline src/app/shared/components/confirm-modal/confirm-modal.component.html]: Here are some uneccessary spaces

File: src/app/shared/components/confirm-modal/confirm-modal.component.ts [hunk 9]
```diff
@@ -0,0 +1,51 @@
+import { Component, OnInit } from '@angular/core';
+import { BsModalRef } from 'ngx-bootstrap/modal/bs-modal-ref.service';
+import { Subject } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+import { Router } from '@angular/router';
+
+@Component({
+  selector: 'app-confirm-modal',
+  templateUrl: './confirm-modal.component.html',
+  styleUrls: ['./confirm-modal.component.scss']
```
Comment (crischiucucu) [inline src/app/shared/components/confirm-modal/confirm-modal.component.ts]: If you don't have any css specified into the scss file please remove it and remove also this option for linking with the scss file.

File: src/app/shared/components/confirm-modal/confirm-modal.component.ts [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+})
+export class ConfirmModalComponent implements OnInit {
+
+  destroy$: Subject<boolean> = new Subject<boolean>();
+
+  message: String;
```
Comment (crischiucucu) [inline src/app/shared/components/confirm-modal/confirm-modal.component.ts]: Same here with the types?

File: src/app/shared/components/confirm-modal/confirm-modal.component.ts [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+
+  service: any;
+  id: string;
+  pageUrl: string;
+
+  constructor(public bsModalRef: BsModalRef,
+    private router: Router) { }
+
+  ngOnInit() {
```
Comment (crischiucucu) [inline src/app/shared/components/confirm-modal/confirm-modal.component.ts]: If you don't have any logic into the method please remove it and remove also the implements OnInit

File: src/app/shared/components/confirm-modal/confirm-modal.component.ts [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+  }
+
+  confirm(): void {
+    this.service.deleteItem(this.id, this.pageUrl).pipe(takeUntil(this.destroy$)).subscribe(
+      suc => {
+        this.message = "Successfully deleted";
+        setTimeout(() => {
+          this.bsModalRef.hide();
+          this.router.navigate([`${this.pageUrl}`]);
```
Comment (crischiucucu) [inline src/app/shared/components/confirm-modal/confirm-modal.component.ts]: As I understood pageUrl is used for the service, but you try to use it into router. What could happen if your api url changes?

File: src/app/shared/services/crud/crud.service.ts [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+import { Observable } from 'rxjs'
+
+@Injectable({
+  providedIn: 'root'
+})
+export class CrudService {
+
+  constructor(private api: BackendApiService) { }
+
+  getNumberOfItems(pageUrl) {
```
Comment (crischiucucu) [inline src/app/shared/services/crud/crud.service.ts]: If you specify on other methods the return type why you don't specify on all methods ? So please add here a return type

**Pull request closed without merging**